demos/python_demos/whiteboard_inpainting_demo/utils/ie_tools.py

```diff
@@ -0,0 +1,99 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import cv2
+import logging as log
+import numpy as np
+import os
+import sys
+
+from openvino.inference_engine import IENetwork, IECore # pylint: disable=import-error,E0611
+
+
+class IEModel:
```

Comment:
The wrappers don't look thoroughly designed, making it (sometimes) not straightforward to understand what is going on.

Reply:
what does exactly confuse you? `IEModel` is a wrapper for inference engine model with two main methods: `preprocess` and `forward`.

Reply:
There are too many unnecessary levels of abstraction. I mean that this wrapper is further wrapped with a wrapper (`MaskRCNN`, `SemanticSegmentation`). And `MaskRCNN` class does not use methods of `IEModel`, but directly calls underlying IE API.

Reply:
Reimplemented with `IEModel` as a base class for both of segmentation models.

Reply:
Looks better now.